crypto/pkcs12/p12_sbag.c

```diff
@@ -212,9 +212,11 @@ PKCS12_SAFEBAG *PKCS12_SAFEBAG_create_pkcs8_encrypt_ex(int pbe_nid,
     EVP_CIPHER *pbe_ciph_fetch = NULL;
     X509_SIG *p8;
 
+    ERR_set_mark();
     pbe_ciph = pbe_ciph_fetch = EVP_CIPHER_fetch(ctx, OBJ_nid2sn(pbe_nid), propq);
     if (pbe_ciph == NULL)
         pbe_ciph = EVP_get_cipherbynid(pbe_nid);
+    ERR_pop_to_mark();
```

Comment:
This should be:

``` C
    if (pbe_ciph == NULL)
        ERR_clear_last_mark();
    else
        ERR_pop_to_mark();
```

Reply:
Not sure it should.. It failed both a fetch and EVP_get_cipherbynid(). If it is supposedly a legacy alg why would it report that it failed the fetch?